packages/ui-kit/src/components/Leaderboard/utils.ts

```diff
@@ -42,20 +45,24 @@ export const getValueWithPrefixAndSufix = (params: {
 }
 
 interface getValueOptions {
-  value: number
+  value: string
   localize?: boolean
 }
 
 const getValue = (options: getValueOptions) => {
   const { value, localize } = options
 
-  const isInteger = Number.isInteger(value)
+  if (getIsNonNumeric(value)) return value
+
+  const numberValue = Number(value)
+
+  const isInteger = Number.isInteger(numberValue)
 
   if (isInteger) {
-    return localize ? value.toLocaleString() : value
+    return localize ? value.toLocaleString() : numberValue
   }
 
-  return localize ? value.toFixed(2).toLocaleString() : value.toFixed(2)
+  return localize ? numberValue.toFixed(2).toLocaleString() : numberValue.toFixed(2)
 }
```

Comment:
Same feedback here as on the Counter variation I left above. In fact, should Leaderboard and Counter just share the same function?

Also, would like to see some unit tests on this function (at a lower-level than the component).

Reply:
Yes! I think both components can use the same function